src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -831,12 +831,34 @@ int muxer_usage(bool is_sdk_present)
 }
 
 // Convert "path" to realpath (merging working dir if needed) and append to "realpaths" out param.
-void append_realpath(const pal::string_t& path, std::vector<pal::string_t>* realpaths)
+void append_probe_realpath(const pal::string_t& path, std::vector<pal::string_t>* realpaths, const pal::string_t& tfm)
 {
-    pal::string_t real = path;
-    if (pal::realpath(&real))
+    pal::string_t probe_path = path;
+
+    if (pal::realpath(&probe_path) && pal::directory_exists(probe_path))
```

Comment:
I tested on my Windows and Linux machines yesterday, and I can create a directory named `{tfm}`.

There aren't many characters you can't use for a name in Linux.  Do we think we are safe with `{}`?  Should we double wrap the curly braces? `{{tfm}}`?

Another option is the `|` pipe character.  You can't use that on Windows, and you have to escape the `|` on Linux.  So I think maybe `|tfm|` and `|arch|` might be good choices.

---
In reply to: [118711887](https://github.com/dotnet/core-setup/pull/2539#discussion_r118711887) [](ancestors = 118711887)